src/hardware/mixer.cpp

```diff
@@ -1693,46 +1694,140 @@ AudioFrame MixerChannel::ApplyCrossfeed(const AudioFrame frame) const
 	return {a.left + b.left, a.right + b.right};
 }
 
-MixerChannel::Sleeper::Sleeper(MixerChannel& c) : channel(c) {}
+void MixerChannel::ConfigureFadeOut(const std::string_view prefs)
+{
+	sleeper.ConfigureFadeOut(prefs);
+}
 
-// Records if samples had a magnitude great than 1. This is a one-way street;
-// once "had_noise" is tripped, it can only be reset after waking up.
-void MixerChannel::Sleeper::Listen(const AudioFrame frame)
+void MixerChannel::Sleeper::ConfigureFadeOut(const std::string_view prefs)
 {
-	if (had_noise) {
+	auto set_wait_and_fade = [&](const int wait_ms, const int fade_ms) {
+		fadeout_or_sleep_after_ms = check_cast<uint16_t>(wait_ms);
+
+		fadeout_decrement_per_ms = 1.0f / static_cast<uint16_t>(fade_ms);
+
+		LOG_MSG("%s: Fadeout enabled (wait %d ms then fade for %d ms)",
+		        channel.GetName().c_str(),
+		        wait_ms,
+		        fade_ms);
+	};
+	// Disable fadeout (default)
+	if (has_false(prefs)) {
+		wants_fadeout = false;
+		return;
+	}
+	// Enable fadeout with defaults
+	if (has_true(prefs)) {
+		set_wait_and_fade(default_wait_ms, default_wait_ms);
+		wants_fadeout = true;
 		return;
 	}
 
-	constexpr auto silence_threshold = 1.0f;
+	// Let the fade out last between 10 ms and 3 seconds.
+	constexpr auto min_fade_ms = 10;
+	constexpr auto max_fade_ms = 3000;
 
-	had_noise = fabsf(frame.left) > silence_threshold ||
-	            fabsf(frame.right) > silence_threshold;
+	// Custom setting in 'WAIT FADE' syntax, where both are milliseconds.
+	if (auto prefs_vec = split(prefs); prefs_vec.size() == 2) {
+		const auto wait_ms = parse_int(prefs_vec[0]);
+		const auto fade_ms = parse_int(prefs_vec[1]);
+		if (wait_ms && fade_ms) {
+			const auto wait_is_valid = (*wait_ms >= min_wait_ms &&
+			                            *wait_ms <= max_wait_ms);
+
+			const auto fade_is_valid = (*fade_ms >= min_fade_ms &&
+			                            *fade_ms <= max_fade_ms);
+
+			if (wait_is_valid && fade_is_valid) {
+				set_wait_and_fade(*wait_ms, *fade_ms);
+				wants_fadeout = true;
+				return;
+			}
+		}
+	}
+	// Otherwise inform the user and disable the fade
+	LOG_WARNING("%s: Invalid custom fadeout definition: '%s'. Must be "
```

Comment:
Ok, checked out the Oxford dictionary, and I think this is the correct usage:

- "fade-out" — that's the noun, so for example "Invalid custom fade-out definition:"
- "fade out" — that's the verb form, e.g., "To fade out stuck notes, use the new `opl_fadeout` setting"
- "fadeout" — not a work 😎 

Could you just grep "fadeout" and "fade out" and update the comments and especially user-facing messages and descriptions accordingly?

Reply:
ok, got 'em :-)